LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,416 @@
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithCustomContainerAsync(request);
+        }
+
+        public async Task CheckAndUploadLinuxContainerImageAsync(string imageSkipToken = null, int pageSize = 10)
+        {
+            do
+            {
+                var ContainerImagesRequest = new ListContainerImagesRequest
+                {
+                    SkipToken = imageSkipToken,
+                    PageSize = pageSize
+                };
+                var ContainerImagesResponse = await PlayFabMultiplayerAPI.ListContainerImagesAsync(ContainerImagesRequest);
+
+                ErrorCheck(ContainerImagesResponse.Error);
+                imageSkipToken = ContainerImagesResponse.Result.SkipToken ?? null;
+                IEnumerable<string> existingImages = ContainerImagesResponse.Result.Images.Where(x => x == settings.ContainerStartParameters.ImageDetails.ImageName);
+                if (!existingImages.Any() && string.IsNullOrEmpty(imageSkipToken))
+                {
+                    //TODO: upload LinuxContainer image API
+                    //This is currently non-existent 
+                    Console.WriteLine("Make sure you have uploaded your Linux container image to Docker before attempting deploy");
+                }
+                
+            } while (!string.IsNullOrEmpty(imageSkipToken));
+        }
+
+        public async Task<PlayFabResult<EmptyResponse>> UploadCertificateAsync(GameCertificateDetails certificate)
+        {
+            X509Certificate2 certCopy = new X509Certificate2(certificate.Path);
+            var uploadCertificateRequest = new UploadCertificateRequest
+            {
+                GameCertificate = new Certificate
+                {
+                    Name = certificate.Name,
+                    Base64EncodedValue = Convert.ToBase64String(certCopy.RawData)
+                    //Password = ""  //Only passwordless certificates supported currently
+                }
+            };
+
+            Console.WriteLine($"Uploading {certificate.Name}...");
+
+            return await PlayFabMultiplayerAPI.UploadCertificateAsync(uploadCertificateRequest);
+        }
+
+        public async Task CheckAndUploadCertificatesAsync(string certSkipToken = null, int pageSize = 10)
+        {
+            List<string> failedCertUploads = new List<string>();
+
+            do
+            {
+                var certificateSummariesRequest = new ListCertificateSummariesRequest
+                {
+                    SkipToken = certSkipToken,
+                    PageSize = pageSize
+                };
+                var certificateSummariesResponse = await PlayFabMultiplayerAPI.ListCertificateSummariesAsync(certificateSummariesRequest);
+
+                ErrorCheck(certificateSummariesResponse.Error);
+
+                certSkipToken = certificateSummariesResponse.Result.SkipToken ?? null;
+
+                foreach (var certificate in settings.GameCertificateDetails)
```

Comment:
I think we need to check if user configured cert information in GameCertificateDetails so we don't need to call ListCertificateSummariesAsync unnecessarily?